src/nvim/edit.c

```diff
@@ -3530,21 +3530,20 @@ int ins_compl_add_tv(typval_T *tv, int dir)
   char_u      *(cptext[CPT_COUNT]);
 
   if (tv->v_type == VAR_DICT && tv->vval.v_dict != NULL) {
-    word = get_dict_string(tv->vval.v_dict, (char_u *)"word", FALSE);
-    cptext[CPT_ABBR] = get_dict_string(tv->vval.v_dict,
-        (char_u *)"abbr", FALSE);
-    cptext[CPT_MENU] = get_dict_string(tv->vval.v_dict,
-        (char_u *)"menu", FALSE);
-    cptext[CPT_KIND] = get_dict_string(tv->vval.v_dict,
-        (char_u *)"kind", FALSE);
-    cptext[CPT_INFO] = get_dict_string(tv->vval.v_dict,
-        (char_u *)"info", FALSE);
-    if (get_dict_string(tv->vval.v_dict, (char_u *)"icase", FALSE) != NULL)
-      icase = get_dict_number(tv->vval.v_dict, (char_u *)"icase");
-    if (get_dict_string(tv->vval.v_dict, (char_u *)"dup", FALSE) != NULL)
-      adup = get_dict_number(tv->vval.v_dict, (char_u *)"dup");
-    if (get_dict_string(tv->vval.v_dict, (char_u *)"empty", FALSE) != NULL)
-      aempty = get_dict_number(tv->vval.v_dict, (char_u *)"empty");
+    word = get_dict_string(tv->vval.v_dict, "word", false);
+    cptext[CPT_ABBR] = get_dict_string(tv->vval.v_dict, "abbr", false);
+    cptext[CPT_MENU] = get_dict_string(tv->vval.v_dict, "menu", false);
+    cptext[CPT_KIND] = get_dict_string(tv->vval.v_dict, "kind", false);
+    cptext[CPT_INFO] = get_dict_string(tv->vval.v_dict, "info", false);
+    if (get_dict_string(tv->vval.v_dict, "icase", false) != NULL) {
```

Comment:
It's because `icase` used to be initialized to `p_ic`: https://github.com/vim/vim/commit/39f05630adc017f00ea551b827fd1d5ef21a0a3a

That was the only case originally, then the pattern just got copied over and over as new cases were added. `get_dict_number` hasn't changed in 12 years.

    git co @~200 && git log -L :get_dict_number:src/eval.c

---

Trivia: looking at the logs I noticed the completion function used to be called `expand_occult`, and "occult completion" was the original name for the completion feature: http://vim.1045645.n5.nabble.com/Occult-completion-td1195634.html